atomisticparsers/gromacs/parser.py

```diff
@@ -652,25 +647,43 @@ def parse_thermodynamic_data(self):
             thermo_data = self.energy_parser
             n_evaluations = self.energy_parser.length
 
-        create_scc = False
-        if n_frames != n_evaluations:
-            self.logger.warning(
-                'Mismatch in number of calculations and number of thermodynamic '
-                'evaluations, will create new sections')
-            create_scc = True
+        system_steps = [frame.step for frame in sec_system]
+        system_times_ps = [frame.time.magnitude * ureg.convert(
+            1.0, frame.time.units, ureg.picosecond) for frame in sec_system]
+        calculation_times_ps = thermo_data.get('Time')
+        calculation_steps = [int((val / time_step).magnitude) for val in calculation_times_ps]
+        calculation_times_ps = calculation_times_ps.magnitude * ureg.convert(
+            1.0, calculation_times_ps.units, ureg.picosecond)
+
+        map_calculation_to_system = {}
```

Comment:
we do not need this map i guess, we can directly create calculation sections here?

Reply:
removed.